worlds/peaks_of_yore/__init__.py

```diff
@@ -0,0 +1,179 @@
+            "Intermediate Book": self.options.enable_intermediate,
+            "Advanced Book": self.options.enable_advanced,
+            "Expert Book": self.options.enable_expert
+        }
+
+        starting_book: Item = self.create_item([*books][self.options.starting_book.value])
+
+        remaining_items: int = len(self.multiworld.get_unfilled_locations(self.player))
+
+        for name, option in books.items():
+            if (not option) or name == starting_book.name or remaining_items <= 0:
+                continue
+            self.multiworld.itempool.append(self.create_item(name))
+            remaining_items -= 1
+
+        for tool in [item for item in full_item_list if item.type == "Tool"]:
+            if remaining_items > 0 and (tool.name != "Barometer" or not self.options.start_with_barometer) \
+                    and (tool.name != "Oil Lamp" or not self.options.start_with_oil_lamp):
+                if tool.name == "Progressive Crampons":
+                    self.multiworld.itempool.append(self.create_item(tool.name))
+                    remaining_items -= 1
+                    if self.options.starting_book.value != StartingBook.option_expert:
+                        self.multiworld.itempool.append(self.create_item(tool.name))
+                        remaining_items -= 1
+                elif tool.name == "Rope Unlock":
+                    if self.options.rope_unlock_mode == RopeUnlockMode.option_early:
+                        self.multiworld.early_items[self.player][tool.name] = 1
+                        self.multiworld.itempool.append(self.create_item(tool.name))
+                        remaining_items -= 1
+                    elif self.options.rope_unlock_mode == RopeUnlockMode.option_normal:
+                        self.multiworld.itempool.append(self.create_item(tool.name))
+                        remaining_items -= 1
+                        # else don't place rope unlock as it will be unlocked with any rope pick-up
+                else:
+                    self.multiworld.itempool.append(self.create_item(tool.name))
+                    remaining_items -= 1
+
+        for rope in [item for item in full_item_list if item.type == "Rope"]:
+            if remaining_items > 0:
+                self.multiworld.itempool.append(self.create_item(rope.name))
+                remaining_items -= 1
+
+        for birdSeed in [item for item in full_item_list if item.type == "Bird Seed"]:
+            if remaining_items > 0:
+                self.multiworld.itempool.append(self.create_item(birdSeed.name))
+                remaining_items -= 1
+
+        for artefact in [item for item in full_item_list if item.type == "Artefact"]:
+            if remaining_items > 0:
+                self.multiworld.itempool.append(self.create_item(artefact.name))
+                remaining_items -= 1
+
+        self.multiworld.itempool += [self.create_item(self.get_filler_item_name()) for _ in range(remaining_items)]
```

Comment:
You can just use the following.

```suggestion
        self.multiworld.itempool += [self.create_filler() for _ in range(remaining_items)]
```